gradle/missing-javadoc.gradle

```diff
@@ -68,6 +72,110 @@ allprojects {
+  project(":distribution:tools:launchers"),
+  project(":distribution:tools:plugin-cli"),
+  project(":doc-tools"),
+  project(":example-plugins:custom-settings"),
+  project(":example-plugins:custom-significance-heuristic"),
+  project(":example-plugins:custom-suggester"),
+  project(":example-plugins:painless-whitelist"),
+  project(":example-plugins:rescore"),
+  project(":example-plugins:rest-handler"),
+  project(":example-plugins:script-expert-scoring"),
+  project(":libs:opensearch-cli"),
+  project(":libs:opensearch-core"),
+  project(":libs:opensearch-dissect"),
+  project(":libs:opensearch-geo"),
+  project(":libs:opensearch-grok"),
+  project(":libs:opensearch-nio"),
+  project(":libs:opensearch-plugin-classloader"),
+  project(":libs:opensearch-secure-sm"),
+  project(":libs:opensearch-ssl-config"),
+  project(":libs:opensearch-x-content"),
+  project(":modules:aggs-matrix-stats"),
+  project(":modules:analysis-common"),
+  project(":modules:geo"),
+  project(":modules:ingest-common"),
+  project(":modules:ingest-geoip"),
+  project(":modules:ingest-user-agent"),
+  project(":modules:lang-expression"),
+  project(":modules:lang-mustache"),
+  project(":modules:lang-painless"),
+  project(":modules:lang-painless:spi"),
+  project(":modules:mapper-extras"),
+  project(":modules:opensearch-dashboards"),
+  project(":modules:parent-join"),
+  project(":modules:percolator"),
+  project(":modules:rank-eval"),
+  project(":modules:reindex"),
+  project(":modules:repository-url"),
+  project(":modules:systemd"),
+  project(":modules:transport-netty4"),
+  project(":plugins:analysis-icu"),
+  project(":plugins:analysis-kuromoji"),
+  project(":plugins:analysis-nori"),
+  project(":plugins:analysis-phonetic"),
+  project(":plugins:analysis-smartcn"),
+  project(":plugins:analysis-stempel"),
+  project(":plugins:analysis-ukrainian"),
+  project(":plugins:discovery-azure-classic"),
+  project(":plugins:discovery-ec2"),
+  project(":plugins:discovery-ec2:qa:amazon-ec2"),
+  project(":plugins:discovery-gce"),
+  project(":plugins:discovery-gce:qa:gce"),
+  project(":plugins:ingest-attachment"),
+  project(":plugins:mapper-annotated-text"),
+  project(":plugins:mapper-murmur3"),
+  project(":plugins:mapper-size"),
+  project(":plugins:repository-azure"),
+  project(":plugins:repository-gcs"),
+  project(":plugins:repository-hdfs"),
+  project(":plugins:repository-s3"),
+  project(":plugins:store-smb"),
+  project(":plugins:transport-nio"),
+  project(":qa:die-with-dignity"),
+  project(":qa:os"),
+  project(":qa:wildfly"),
+  project(":rest-api-spec"),
+  project(":server"),
+  project(":test:external-modules:test-delayed-aggs"),
+  project(":test:fixtures:azure-fixture"),
+  project(":test:fixtures:gcs-fixture"),
+  project(":test:fixtures:hdfs-fixture"),
+  project(":test:fixtures:old-elasticsearch"),
+  project(":test:fixtures:s3-fixture"),
+  project(":test:framework"),
+  project(":test:logger-usage")
+]) {
```

Comment:
This is basically everything, especially with `:server`. Feels like we should machine-generate JavaDoc first, or divide and conquer manually adding it, or something like that before enabling this.

Reply:
Thanks for reviewing @dblock. I agree we should add more javadocs, and divide and conquer seems to be  the right approach, given the extent. However to your point on publishing, we do have machine generated javadocs today on opensearch [website](https://opensearch.org/docs/javadocs/). Enabling this check will not break the javadoc task (used for publishing these docs on website), instead it would ensure that the modules that have been fixed (like client:rest #685) do not break again.

Reply:
Open an issue for missing JavaDoc when merging?

Reply:
> Open an issue for missing JavaDoc when merging and link to this conversation?

Already there :) #221